editorextensions/codebeamer-custom-ui/modal/src/api/lucidGateway.ts

```diff
@@ -130,3 +137,22 @@ export function startImport(id: number, items: number) {
 export function closeModal() {
 	window.parent.postMessage({ action: 'closeModal', payload: true }, '*');
 }
+
+export function handleCardBlocks(callback: (arg0: any) => void) {
+	const getCardBlocks = () => {
+		window.parent.postMessage(
+			{ action: 'getCardBlocks', payload: {} },
+			'*'
+		);
+	};
+
+	const addMessageListener = () => {
+		window.addEventListener('message', (e) => {
+			const data = JSON.parse(e.data);
+			callback(data);
+		});
+	};
```

Comment:
Setting up this de facto global event listener ( `addMessageListener`) in one specific function isn't scalable very well.  
The listener should either be set up procedurally, or upon instantiation of the (not currently existing) class `LucidGateway`.  

Then you might want to just expose an API to _register_  callbacks that fire whenever we _do_ get such a message. See also [observer pattern](https://refactoring.guru/design-patterns/observer).

Reply:
Created a MessageHandler class that does this